### PR TITLE
Implement egress only internet gateway nuke

### PR DESCRIPTION
**Pull request by hongil0316**
## Description

Partially Fixing https://github.com/gruntwork-io/cloud-nuke/issues/476

## TODOs

Read the [Gruntwork contribution guidelines](https://gruntwork.notion.site/Gruntwork-Coding-Methodology-02fdcd6e4b004e818553684760bf691e).

- [ ] Update the docs.
- [ ] Run the relevant tests successfully, including pre-commit checks.
- [ ] Ensure any 3rd party code adheres with our [license policy](https://www.notion.so/gruntwork/Gruntwork-licenses-and-open-source-usage-policy-f7dece1f780341c7b69c1763f22b1378) or delete this line if its not applicable.
- [ ] Include release notes. If this PR is backward incompatible, include a migration guide.
- [ ] Attention Grunts - if this PR adds support for a new resource, ensure the `nuke_sandbox` and `nuke_phxdevops` jobs in `.circleci/config.yml` have been updated with appropriate exclusions (either directly in the job or via the `.circleci/nuke_config.yml` file) to prevent nuking IAM roles, groups, resources, etc that are important for the test accounts.

## Release Notes (draft)

Added / Removed / Updated [X].

### CODE REVIEW

File: aws/resources/ec2_egress_only_igw.go [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+	}
+
+	logging.Debugf("Deleting all Egress only internet gateway in region %s", egigw.Region)
+	var deletedList []*string
+
+	for _, id := range ids {
+
+		if nukable, err := egigw.IsNukable(*id); !nukable {
+			logging.Debugf("[Skipping] %s nuke because %v", *id, err)
+			continue
```
Comment (denis256) [inline aws/resources/ec2_egress_only_igw.go:121]: Is it fine that errors are ignored?

File: aws/resources/ec2_egress_only_igw.go [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+				continue
+			}
+		}
+
+		if shouldIncludeEgressOnlyInternetGateway(igw, firstSeenTime, configObj) {
+			result = append(result, igw.EgressOnlyInternetGatewayId)
+		}
+	}
+
+	// chgecking the nukable permissions
```
Comment (denis256) [inline aws/resources/ec2_egress_only_igw.go:96]: `chgecking` :thinking:

File: aws/resources/ec2_egress_only_igw_test.go [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+			configObj: config.ResourceType{
+				ExcludeRule: config.FilterRule{
+					TimeBefore: aws.Time(now.Add(1)),
+				}},
+			expected: []string{gateway2},
+		},
+	}
+	for name, tc := range tests {
+		t.Run(name, func(t *testing.T) {
+			names, err := object.getAll(context.Background(), config.Config{
```
Comment (denis256) [inline aws/resources/ec2_egress_only_igw_test.go:113]: I suspect t.Parallel() should be invoked

File: aws/resources/ec2_egress_only_igw.go (hunk 1 above)
Comment (hongil0316) [reply aws/resources/ec2_egress_only_igw.go:123]: Explain in the comment. Yeah it is safe to ignore in this cas.e

File: aws/resources/ec2_egress_only_igw_test.go (hunk 3 above)
Comment (hongil0316) [reply aws/resources/ec2_egress_only_igw_test.go:114]: Added.

**Review by denis256 — APPROVED** [review]

**Pull request merged**